### PR TITLE
multichain module docs

### PR DESCRIPTION
**Pull request by livingrockrises**

### CODE REVIEW

File: docs/Modules/multichain.mdx [hunk 1]
```diff
@@ -1,22 +1,153 @@
...
 
-This module acts as an extention of the [ECDSA Module](/Modules/ecdsa). The added benefit of this module is you now have the ability to use a single signer accross multiple blockchain networks. This allows you now to completely abstract away the chain that the user is interacting with.
+This module acts as an extention of the [ECDSA Module](/Modules/ecdsa). The added benefit of this module is you now have the ability to sign once and authorize any amount of userOps accross multiple blockchain networks.
 
-## Installation
+### What is the Multichain Validation Module? 🤔
+
+This module allows **Externally Owned Accounts (EOAs)** to authorize and sign user operations (UserOps) for Biconomy Smart Accounts on Multiple chains with single ECDSA signature. 
+
+`Benefits`
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:18]:
```suggestion
**Benefits**
```

File: docs/Modules/multichain.mdx [hunk 2]
```diff
@@ -1,22 +1,153 @@
...
+- **ECDSA Signature Scheme**: Utilizes the ECDSA secp256r1 curve for secure signing.
+- **Flexible Signer Authentication**: Supports various signer solutions like Privy, Fireblocks, Arcana Auth, Capsule, Lit protocol, Turnkey, Web3Auth, Particle, Magic, Portal etc
+- **EIP-1271 Compliance**: Allowing Smart Accounts to sign Ethereum messages for logging into dApps.
+- **Sign once, Execute on multiple chains**
+
+#### Use Cases 🌟
+
+Enables use cases which require several actions to be authorized for several chains with just one signature required from user.
+
+1. **Multichain intent for example exit all my native tokens for usdc
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:37]: double stars can be removed.

File: docs/Modules/multichain.mdx [hunk 3]
```diff
@@ -1,22 +1,153 @@
...
+    ) external view virtual override returns (uint256) {
+
+       // userOp.signature contains both the signature and the validation module's address
+
+        (bytes memory moduleSignature, ) = abi.decode(
+            userOp.signature,
+            (bytes, address)
+        );
+
+
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:73]: can we remove extra new lines and spaces from the code

File: docs/Modules/multichain.mdx [hunk 4]
```diff
@@ -1,22 +1,153 @@
...
+```
+
+All the other methods are inherited from [ECDSAOwnershipRegistryModule](https://github.com/bcnmy/scw-contracts/blob/main/contracts/smart-account/modules/EcdsaOwnershipRegistryModule.sol)
 
-import Tabs from "@theme/Tabs";
-import TabItem from "@theme/TabItem";
+
+#### References
+
+1.**[Source code](https://github.com/bcnmy/scw-contracts/blob/develop/contracts/smart-account/modules/MultichainECDSAValidator.sol)**
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:132]:
```suggestion
1. **[Source code](https://github.com/bcnmy/scw-contracts/blob/develop/contracts/smart-account/modules/MultichainECDSAValidator.sol)**
```

File: docs/Modules/multichain.mdx [hunk 5]
```diff
@@ -1,22 +1,153 @@
...
+3. Every userOpHash is now a leaf of a Merkle Tree.
+4. Instead of signing several userOpHash separately, user signs one Merkle Root.
+5. Now we can reliably prove on-chain that a userOp is a leaf of a tree whose root has been signed by an authorized user.
+
+Now user can authorize any amount of userOps with just one signature over the Merkle Root of those userOps
+
+
+## Smart Contract Deep Dive 🛠️
+
+You can send a multi-chain signature to send different userops on different chains and be validated by this module. if plain ecdsa signature(65 bytes) is sent then the module assumes it’s form as standard ECDSA validation module
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:55]: can we make it more clearer?

File: docs/Modules/multichain.mdx [hunk 6]
```diff
@@ -34,49 +165,117 @@ yarn add @biconomy/modules
...
+### Creating a Signer 🗝️
+
+```javascript
+import { ethers } from "ethers";
+
+const provider = new ethers.providers.JsonRpcProvider("[RPC_Endpoint]");
+const signer = new ethers.Wallet("[Private_Key]", provider);
+```
+
+### Integration - Importing and Initializing Multichain Validation Module 🌟
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:184]:
```suggestion
### Importing and Initializing Multichain Validation Module
```

File: docs/Modules/multichain.mdx [hunk 7]
```diff
@@ -34,49 +165,117 @@ yarn add @biconomy/modules
...
+```
+
+:::info
+If Smart Account is not deployed yet then it makes sense to pass multiChainModule as defaultValidationModule. That was it would get enabled by default and has all Ecdsa Module (popular go-to module) capabilities. 
+:::
+
+
+
+
+### Conclusion
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:277]:
```suggestion
## Conclusion
```

File: docs/Modules/multichain.mdx [hunk 8]
```diff
@@ -34,49 +165,117 @@ yarn add @biconomy/modules
...
+    rpcUrl: // Polygon Mumbai Rpc
     entryPointAddress: DEFAULT_ENTRYPOINT_ADDRESS,
-    defaultValidationModule: multiChainModule,
+    defaultValidationModule: multiChainModule, // this can also be ecdsaModule instance
     activeValidationModule: multiChainModule
   });
 
 ```
+
+### Integration - Signing and Sending Multichain userops🌟
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:231]:
```suggestion
### Signing and Sending Multichain userOps
```

File: docs/Modules/multichain.mdx [hunk 9]
```diff
@@ -1,22 +1,153 @@
...
+2.**[Merkle Trees](https://www.simplilearn.com/tutorials/blockchain-tutorial/merkle-tree-in-blockchain)**
+
+
+## How to Guide - Biconomy SDK 📘 
+
+Integrate the Multichain Validation Module into Biconomy Smart Accounts using the SDK.
+
+
+
+### Setup and Installation 🛠️
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:143]:
```suggestion
### Setup and Installation
```

File: docs/Modules/multichain.mdx [hunk 10]
```diff
@@ -34,49 +165,117 @@ yarn add @biconomy/modules
...
-pnpm add @biconomy/modules
+pnpm add @biconomy/modules ethers
 
 ```
 
 </TabItem>
 </Tabs>
 
-## Integration
+### Creating a Signer 🗝️
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:175]:
```suggestion
### Creating a Signer
```

File: docs/Modules/multichain.mdx [hunk 11]
```diff
@@ -34,49 +165,117 @@ yarn add @biconomy/modules
...
 });
 ```
 
-Once initialized it can be passed to the smart account [create method](/Account/methods/create) to create a smart account.
+Once initialized it can be passed to the smart account [create method](/Account/methods/create) to create instances of the smart account.
+Check [Paymaster integration](/Paymaster/integration) and [Bundler integration](/Bundler/integration) sections to create instances of them.
 
 Here is how you can set up Smart Account instances accross multiple chains:
 
+
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:204]:
```suggestion
```

File: docs/Modules/multichain.mdx [hunk 12]
```diff
@@ -34,49 +165,117 @@ yarn add @biconomy/modules
...
 ```ts
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:206]:
```suggestion
```

File: docs/Modules/multichain.mdx [hunk 13]
```diff
@@ -1,22 +1,153 @@
...
+
+### How does it work? 🤔
+
+1. We take all the userOps which we need to sign.
+2. We get userOpHash from those userOps.
+3. Every userOpHash is now a leaf of a Merkle Tree.
+4. Instead of signing several userOpHash separately, user signs one Merkle Root.
+5. Now we can reliably prove on-chain that a userOp is a leaf of a tree whose root has been signed by an authorized user.
+
+Now user can authorize any amount of userOps with just one signature over the Merkle Root of those userOps
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:50]:
```suggestion
With this, Users can authorize any number of userOps with signing merkle root of those userOps Hashes.
```

File: docs/Modules/multichain.mdx [hunk 14]
```diff
@@ -1,22 +1,153 @@
...
+Enables use cases which require several actions to be authorized for several chains with just one signature required from user.
+
+1. **Multichain intent for example exit all my native tokens for usdc
+2. **Allows a dApp to require just one signature from their user to configure & deploy smart accounts on multiple chains and delegate certain actions with permission via session keys on all those chains
+3. **Cross chain state sync actions e.g transfer ownership of SA, change ens, similar actions on protocol that is deployed on multiple chains
+4. **Onboard user on x chains by deploying their smart account + some action like register in raffle 
+
+### How does it work? 🤔
+
+1. We take all the userOps which we need to sign.
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:44]: can we rephrase these steps
It involves collecting the set of userOps for signing, 
each userOp Hash is a leaf in a Merkle Tree.
Instead of signing individual hashes, user signs the Merkle Root.
This signed Merkle Root serves as on-chain evidence, proving any userOp within the tree is authorized.

File: docs/Modules/multichain.mdx [hunk 15]
```diff
@@ -34,49 +165,117 @@ yarn add @biconomy/modules
...
+:::info
+If Smart Account is not deployed yet then it makes sense to pass multiChainModule as defaultValidationModule. That was it would get enabled by default and has all Ecdsa Module (popular go-to module) capabilities. 
+:::
+
+
+
+
+### Conclusion
+
+It reduces user friction for dApps which for example want to quickly and seamlessly deploy and configure Smart Accounts on several chains or issue session keys with the permissions that vary from chain to chain.
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:279]: can we simplify this?

File: docs/Modules/multichain.mdx [hunk 16]
```diff
@@ -1,22 +1,153 @@
...
+description: "Overview of Multichain Validation Module with ECDSA Ownership"
 ---
 
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
 # Multichain Validation Module
 
-This module acts as an extention of the [ECDSA Module](/Modules/ecdsa). The added benefit of this module is you now have the ability to use a single signer accross multiple blockchain networks. This allows you now to completely abstract away the chain that the user is interacting with.
+This module acts as an extention of the [ECDSA Module](/Modules/ecdsa). The added benefit of this module is you now have the ability to sign once and authorize any amount of userOps accross multiple blockchain networks.
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:12]:
```suggestion
This module acts as an extention of the [ECDSA Module](/Modules/ecdsa). The added benefit of this module is you now have the ability to sign once and authorize any number of userOps across multiple blockchain networks.
```

File: docs/Modules/multichain.mdx [hunk 17]
```diff
@@ -1,22 +1,153 @@
...
+#### Key Functions 🔐
+
+- **ECDSA Signature Scheme**: Utilizes the ECDSA secp256r1 curve for secure signing.
+- **Flexible Signer Authentication**: Supports various signer solutions like Privy, Fireblocks, Arcana Auth, Capsule, Lit protocol, Turnkey, Web3Auth, Particle, Magic, Portal etc
+- **EIP-1271 Compliance**: Allowing Smart Accounts to sign Ethereum messages for logging into dApps.
+- **Sign once, Execute on multiple chains**
+
+#### Use Cases 🌟
+
+Enables use cases which require several actions to be authorized for several chains with just one signature required from user.
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:35]:
```suggestion
```

Comment (himanshugarg06) [reply docs/Modules/multichain.mdx:35]: This is repetitive and redundant

File: docs/Modules/multichain.mdx [hunk 18]
```diff
@@ -34,49 +165,117 @@ yarn add @biconomy/modules
...
+  const userOpResponse1 = await baseAccount.sendSignedUserOp(returnedOps[0] as any);
+  // You can also just wait for transaction hash by waitForTxHash()
+  const transactionDetails1 = await userOpResponse1.wait();
+
+  const userOpResponse2 = await polygonAccount.sendSignedUserOp(returnedOps[1] as any);
+  const transactionDetails2 = await userOpResponse2.wait();
+```
+
+:::info
+If Smart Account is not deployed yet then it makes sense to pass multiChainModule as defaultValidationModule. That was it would get enabled by default and has all Ecdsa Module (popular go-to module) capabilities.
```
Comment (himanshugarg06) [inline docs/Modules/multichain.mdx:271]:
```suggestion
If Smart Account has not been deployed yet then it makes sense to pass multiChainModule as defaultValidationModule. This ensures its default activation and has all the capabilities of the ECDSA Module (popular go-to module). 
```

**Review by himanshugarg06 — CHANGES_REQUESTED** [review]

File: docs/Modules/multichain.mdx (hunk 16 above)
Comment (Aboudjem) [inline docs/Modules/multichain.mdx:12]: extention -> extension

**Review by Aboudjem — CHANGES_REQUESTED** [review]

File: docs/Modules/multichain.mdx [hunk 19]
```diff
@@ -1,22 +1,153 @@
 ---
+sidebar_label: "Multichain Validation Module"
```
Comment (NunoFilipeSantos) [inline docs/Modules/multichain.mdx:2]:
```suggestion
sidebar_label: "Multichain Validation"
```

File: docs/Modules/multichain.mdx (hunk 16 above)
Comment (NunoFilipeSantos) [inline docs/Modules/multichain.mdx:12]:
```suggestion
This module acts as an extension of the [ECDSA Module](/Modules/ecdsa). The added benefit of this module is you now have the ability to sign once and authorize any amount of userOps across multiple blockchain networks.
```
Please use a grammar checker.

File: docs/Modules/multichain.mdx (hunk 2 above)
Comment (NunoFilipeSantos) [inline docs/Modules/multichain.mdx:37]:
```suggestion
1. **Multichain intent for example exit all my native tokens for USDC
```

File: docs/Modules/multichain.mdx [hunk 20]
```diff
@@ -1,22 +1,153 @@
...
+- **Sign once, Execute on multiple chains**
+
+#### Use Cases 🌟
+
+Enables use cases which require several actions to be authorized for several chains with just one signature required from user.
+
+1. **Multichain intent for example exit all my native tokens for usdc
+2. **Allows a dApp to require just one signature from their user to configure & deploy smart accounts on multiple chains and delegate certain actions with permission via session keys on all those chains
+3. **Cross chain state sync actions e.g transfer ownership of SA, change ens, similar actions on protocol that is deployed on multiple chains
+4. **Onboard user on x chains by deploying their smart account + some action like register in raffle
```
Comment (NunoFilipeSantos) [inline docs/Modules/multichain.mdx:40]:
```suggestion
2. **Allows a dApp to require just one signature from their user to configure & deploy smart accounts on multiple chains and delegate certain actions with permission via session keys on all those chains
3. **Cross chain state sync actions e.g transfer ownership of a smart account, change ENS, similar actions on protocol that is deployed on multiple chains
4. **Onboard a user on x chains by deploying their smart account and some action like register in a raffle 
```

"similar actions on protocol that is deployed on multiple chains" Can you rephrase this sentence it is unclear to me.

File: docs/Modules/multichain.mdx [hunk 21]
```diff
@@ -1,22 +1,153 @@
...
+3. **Cross chain state sync actions e.g transfer ownership of SA, change ens, similar actions on protocol that is deployed on multiple chains
+4. **Onboard user on x chains by deploying their smart account + some action like register in raffle 
+
+### How does it work? 🤔
+
+1. We take all the userOps which we need to sign.
+2. We get userOpHash from those userOps.
+3. Every userOpHash is now a leaf of a Merkle Tree.
+4. Instead of signing several userOpHash separately, user signs one Merkle Root.
+5. Now we can reliably prove on-chain that a userOp is a leaf of a tree whose root has been signed by an authorized user.
```
Comment (NunoFilipeSantos) [inline docs/Modules/multichain.mdx:48]:
```suggestion
2. We get userOpHash from those userOps.
3. Every userOpHash is now a leaf of a Merkle Tree.
4. Instead of signing several userOpHashes separately, the user signs one Merkle Root.
5. Now we can reliably prove on-chain that a userOp is a leaf of a tree whose root has been signed by an authorized user.
```

File: docs/Modules/multichain.mdx (hunk 13 above)
Comment (NunoFilipeSantos) [inline docs/Modules/multichain.mdx:50]:
```suggestion
Now the user can authorize any amount of userOps (Merkle Tree leafs) with just one signature over the Merkle Tree Root.
```

File: docs/Modules/multichain.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (NunoFilipeSantos) [reply docs/Modules/multichain.mdx:55]: - userops -> userOps
- "ECDSA signature(65 bytes)"-> ECDSA signature (65 bytes)

File: docs/Modules/multichain.mdx (hunk 3 above)
Comment (NunoFilipeSantos) [reply docs/Modules/multichain.mdx:73]: [Solidity Style Guide](https://docs.soliditylang.org/en/latest/style-guide.html#blank-lines)

File: docs/Modules/multichain.mdx (hunk 18 above)
Comment (NunoFilipeSantos) [reply docs/Modules/multichain.mdx:271]: If "the" Smart Account.

File: docs/Modules/multichain.mdx [hunk 22]
```diff
@@ -34,49 +165,117 @@ yarn add @biconomy/modules
...
+:::
+
+
+
+
+### Conclusion
+
+It reduces user friction for dApps which for example want to quickly and seamlessly deploy and configure Smart Accounts on several chains or issue session keys with the permissions that vary from chain to chain.
+
+"Mutlichain module allows us to use sessions keys and setup sessions on multiple chains with single wallet signature." - [Rage Trade](https://www.rage.trade/)
```
Comment (NunoFilipeSantos) [inline docs/Modules/multichain.mdx:281]:
```suggestion
"Multichain module allows us to use session keys and setup sessions on multiple chains with single wallet signature." - [Rage Trade](https://www.rage.trade/)
```

**Review by NunoFilipeSantos — CHANGES_REQUESTED** [review]
multiple typos.

File: docs/Modules/multichain.mdx (hunk 19 above)
[the remark this answers is not in the document]
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:2]: why??

[the remark this answers is not in the document]
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:2]: hmm ok

File: docs/Modules/multichain.mdx (hunk 17 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:35]: I think there's no hard keeping it to set context for example use cases

File: docs/Modules/multichain.mdx (hunk 20 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:40]: if uniswap is on 3 chains then make a swap USDC to ETH on all chains

File: docs/Modules/multichain.mdx (hunk 14 above)
Comment (NunoFilipeSantos) [reply docs/Modules/multichain.mdx:44]: Agree.

File: docs/Modules/multichain.mdx (hunk 20 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:40]: added suggestions

File: docs/Modules/multichain.mdx (hunk 14 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:44]: check the changes

Comment (livingrockrises) [reply docs/Modules/multichain.mdx:44]: ### How does it work? 🤔

1. Collecting the set of userOps for signing
2. Calculate userOpHash
3. Every userOpHash is now a leaf of a Merkle Tree.
4. Instead of signing individual hashes, user signs the Merkle Root.

Now this signed Merkle Root serves as on-chain evidence, proving any userOp within the tree is authorized. User can authorize any amount of userOps with just one signature over the Merkle Root of those userOps

File: docs/Modules/multichain.mdx (hunk 21 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:48]: done

File: docs/Modules/multichain.mdx (hunk 13 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:50]: done

File: docs/Modules/multichain.mdx (hunk 5 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:55]: it's explained in smart contract section below. what do you propose?

Comment (livingrockrises) [reply docs/Modules/multichain.mdx:55]: doing this

"You have the option to transmit a multi-chain signature, enabling the sending of various user operations on distinct chains. This signature is then validated by the module. Alternatively, if a plain ECDSA signature (65 bytes) is sent, the module interprets it as a standard ECDSA validation."

[the remark this answers is not in the document]
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:55]: updated

File: docs/Modules/multichain.mdx (hunk 3 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:73]: I have copied this from code in scw repo which uses solidity linter

File: docs/Modules/multichain.mdx (hunk 4 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:132]: done

File: docs/Modules/multichain.mdx [hunk 23]
```diff
@@ -1,22 +1,153 @@
...
+
+2.**[Merkle Trees](https://www.simplilearn.com/tutorials/blockchain-tutorial/merkle-tree-in-blockchain)**
+
+
+## How to Guide - Biconomy SDK 📘 
+
+Integrate the Multichain Validation Module into Biconomy Smart Accounts using the SDK.
+
+
+
```
[the remark this answers is not in the document]
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:142]: ok

File: docs/Modules/multichain.mdx (hunk 9 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:143]: done

File: docs/Modules/multichain.mdx (hunk 10 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:175]: done

File: docs/Modules/multichain.mdx (hunk 6 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:184]: done

File: docs/Modules/multichain.mdx (hunk 8 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:231]: done

File: docs/Modules/multichain.mdx (hunk 18 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:271]: done

File: docs/Modules/multichain.mdx (hunk 7 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:277]: ok

File: docs/Modules/multichain.mdx (hunk 15 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:279]: It makes it easier for dApps to deploy and set up Smart Accounts across multiple chains or issue session keys with different permissions for each chain, reducing user difficulties.

Comment (livingrockrises) [reply docs/Modules/multichain.mdx:279]: done

File: docs/Modules/multichain.mdx (hunk 22 above)
[the remark this answers is not in the document]
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:281]: nice catch

Comment (livingrockrises) [reply docs/Modules/multichain.mdx:281]: done

**Comment by livingrockrises** [comment]
updated PR. re requested review for approval. let's merge it this year!

File: docs/Modules/multichain.mdx [hunk 24]
```diff
@@ -1,22 +1,136 @@
...
+description: "Overview of Multichain Validation Module with ECDSA Ownership"
 ---
 
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
 # Multichain Validation Module
 
-This module acts as an extention of the [ECDSA Module](/Modules/ecdsa). The added benefit of this module is you now have the ability to use a single signer accross multiple blockchain networks. This allows you now to completely abstract away the chain that the user is interacting with.
+This module acts as an extention of the [ECDSA Module](/Modules/ecdsa). The added benefit of this module is you now have the ability to sign once and authorize any number of userOps across multiple blockchain networks.
```
Comment (NunoFilipeSantos) [inline docs/Modules/multichain.mdx:12]:
```suggestion
This module acts as an extension of the [ECDSA Module](/Modules/ecdsa). The added benefit of this module is you now have the ability to sign once and authorize any number of userOps across multiple blockchain networks.
```

File: docs/Modules/multichain.mdx [hunk 25]
```diff
@@ -1,22 +1,136 @@
...
+- **Sign once, Execute on multiple chains**
+
+#### Use Cases 🌟
+
+Enables use cases which require several actions to be authorized for several chains with just one signature required from user.
+
+1. Multichain intent for example exit all my native tokens for USDC
+2. Allows a dApp to require just one signature from their user to configure & deploy smart accounts on multiple chains and delegate certain actions with permission via session keys on all those chains
+3. Cross chain state sync actions e.g transfer ownership of a smart account, change ENS, similar actions on protocol that is deployed on multiple chains
+4. Onboard a user on x chains by deploying their smart account and some action like register in a raffle
```
Comment (NunoFilipeSantos) [inline docs/Modules/multichain.mdx:40]:
```suggestion
1. Multichain intent for example exit all my native tokens for USDC
2. Allows a dApp to require just one signature from their user to configure & deploy smart accounts on multiple chains and delegate certain actions with permission via session keys on all those chains
3. Cross-chain state sync actions e.g transfer ownership of a smart account, change ENS, similar actions on a protocol that are deployed on multiple chains
4. Onboard a user on x chains by deploying their smart account and some action like registering in a raffle 
```

**Comment by livingrockrises** [comment]
> Typos and articles missing.

committed suggestions for typos.
which article?

File: docs/Modules/multichain.mdx [hunk 26]
```diff
@@ -1,22 +1,136 @@
...
-This module acts as an extention of the [ECDSA Module](/Modules/ecdsa). The added benefit of this module is you now have the ability to use a single signer accross multiple blockchain networks. This allows you now to completely abstract away the chain that the user is interacting with.
+This module acts as an extension of the [ECDSA Module](/Modules/ecdsa). The added benefit of this module is you now have the ability to sign once and authorize any number of userOps across multiple blockchain networks.
+
+### What is the Multichain Validation Module? 🤔
+
+This module allows **Externally Owned Accounts (EOAs)** to authorize and sign user operations (UserOps) for Biconomy Smart Accounts on Multiple chains with a single ECDSA signature. 
+
+**Benefits**
+- significantly improves UX for deploying and setting up Smart Accounts on several chains 
+- significantly reduces user friction for multichain operations.
```
Comment (Aboudjem) [inline docs/Modules/multichain.mdx:20]:
```suggestion
- Significantly improves UX for deploying and setting up Smart Accounts on several chains 
- Significantly reduces user friction for multichain operations.
```

File: docs/Modules/multichain.mdx [hunk 27]
```diff
@@ -1,22 +1,136 @@
...
+
+:::tip
+You can also use this module as the **default module** just like the ECDSA Validation Module, as it has all its capabilities for sending ops on the particular chain.
```
Comment (Aboudjem) [inline docs/Modules/multichain.mdx:23]:
```suggestion
You can also use this module as the **default module** just like the ECDSA Validation Module, as it has all its capabilities for sending userOps on the particular chain.
```

File: docs/Modules/multichain.mdx [hunk 28]
```diff
@@ -1,22 +1,136 @@
...
+3. Cross-chain state sync actions e.g transfer ownership of a smart account, change ENS, similar actions on a protocol that are deployed on multiple chains
+4. Onboard a user on x chains by deploying their smart account and some action like registering in a raffle 
+
+### How does it work? 🤔
+
+1. Collecting the set of userOps for signing
+2. Calculate userOpHash
+3. Every userOpHash is now a leaf of a Merkle Tree.
+4. Instead of signing individual hashes, the user signs the Merkle Root.
+
```
Comment (Aboudjem) [inline docs/Modules/multichain.mdx:48]:
```suggestion
The module streamlines operations through a Merkle Tree structure:
- **Gather User Operations**: Compiles various user operations (UserOps) for signature.
- **UserOp Hashing**: Each UserOp is hashed, turning them into individual leaves of a Merkle Tree.
- **Merkle Root Signing**: Instead of signing each hash separately, users sign the Merkle Tree's root, encompassing all operations.
```

File: docs/Modules/multichain.mdx [hunk 29]
```diff
@@ -1,22 +1,136 @@
...
+- **Sign once, Execute on multiple chains**
+
+#### Use Cases 🌟
+
+Enables use cases which require several actions to be authorized for several chains with just one signature required from user.
+
+1. Multichain intent for example exit all my native tokens for USDC
+2. Allows a dApp to require just one signature from their user to configure & deploy smart accounts on multiple chains and delegate certain actions with permission via session keys on all those chains
+3. Cross-chain state sync actions e.g transfer ownership of a smart account, change ENS, similar actions on a protocol that are deployed on multiple chains
+4. Onboard a user on x chains by deploying their smart account and some action like registering in a raffle
```
Comment (Aboudjem) [inline docs/Modules/multichain.mdx:40]:
```suggestion
Here's a rundown of its capabilities:

- **Streamlined Token Management**: For instance, users can convert all native tokens to USDC across different chains effortlessly.
- **dApp Integration**: dApps can now request just one signature from users to set up smart accounts on multiple chains. This includes delegating permissions via session keys on each chain.
- **Cross-Chain Coordination**: Facilitates synchronization tasks like transferring smart account ownership or updating ENS across various chain-specific protocols.
- **User Onboarding Simplified**: Enables quick user onboarding on several chains, such as deploying smart accounts and engaging in activities like raffle registration, using one operation.
```

File: docs/Modules/multichain.mdx [hunk 30]
```diff
@@ -1,22 +1,136 @@
...
+2. Calculate userOpHash
+3. Every userOpHash is now a leaf of a Merkle Tree.
+4. Instead of signing individual hashes, the user signs the Merkle Root.
+
+Now this signed Merkle Root serves as on-chain evidence, proving any userOp within the tree is authorized. With this, Users can authorize any number of userOps with just one signature over the Merkle Tree Root.
+
+
+## Smart Contract Deep Dive 🛠️
+
+You have the option to transmit a multi-chain signature, enabling the sending of various user operations on distinct chains. This signature is then validated by the module. Alternatively, if a plain ECDSA signature (65 bytes) is sent, the module interprets it as a standard ECDSA validation.
```
Comment (Aboudjem) [inline docs/Modules/multichain.mdx:54]:
```suggestion
Offers the ability to transmit a **multi-chain signature**, facilitating various user operations across distinct chains. The module validates this signature accordingly.

:::note
If you send a normal **ECDSA signature** (65 bytes), the module uses standard ECDSA validation.
:::
```

File: docs/Modules/multichain.mdx [hunk 31]
```diff
@@ -1,22 +1,136 @@
...
+
+#### User Operation Validation (`validateUserOp`) 🛡️
+
+```solidity
```
Comment (Aboudjem) [inline docs/Modules/multichain.mdx:58]:
```suggestion
```typescript
```

Docuaurus doesn't properly handle solidity and lacks ofcode colors. Using Typescript here would improve it

File: docs/Modules/multichain.mdx [hunk 32]
```diff
@@ -34,49 +148,106 @@ yarn add @biconomy/modules
...
 ```
+
+:::info
+If the Smart Account has not been deployed yet then it makes sense to pass multiChainModule as defaultValidationModule. This ensures its default activation and has all the capabilities of the ECDSA Module (popular go-to module). 
+:::
+
+## Conclusion
+It makes it easier for dApps to deploy and set up Smart Accounts across multiple chains or issue session keys with different permissions for each chain, reducing user difficulties.
+
+"Multichain module allows us to use session keys and setup sessions on multiple chains with single wallet signature." - [Rage Trade](https://www.rage.trade/)
```
Comment (Aboudjem) [inline docs/Modules/multichain.mdx:253]:
```suggestion
> Multichain module allows us to use session keys and setup sessions on multiple chains with single wallet signature.
>
> — [Rage Trade](https://www.rage.trade/)
```

File: docs/Modules/multichain.mdx [hunk 33]
```diff
@@ -34,49 +148,106 @@ yarn add @biconomy/modules
...
+  const userOpResponse1 = await baseAccount.sendSignedUserOp(returnedOps[0] as any);
+  // You can also just wait for transaction hash by waitForTxHash()
+  const transactionDetails1 = await userOpResponse1.wait();
+
+  const userOpResponse2 = await polygonAccount.sendSignedUserOp(returnedOps[1] as any);
+  const transactionDetails2 = await userOpResponse2.wait();
 ```
+
+:::info
+If the Smart Account has not been deployed yet then it makes sense to pass multiChainModule as defaultValidationModule. This ensures its default activation and has all the capabilities of the ECDSA Module (popular go-to module).
```
Comment (Aboudjem) [inline docs/Modules/multichain.mdx:247]:
```suggestion
When the Smart Account isn't yet deployed, setting **multiChainModule** as the default validation module is beneficial. This activates it by default and retains all functions of the **ECDSA Module**, a popular go-to module.
```

File: docs/Modules/multichain.mdx (hunk 31 above)
Comment (NunoFilipeSantos) [reply docs/Modules/multichain.mdx:58]: We _could_ add it, since Prism already supports it. https://docusaurus.io/docs/markdown-features/code-blocks#supported-languages

https://prismjs.com/#supported-languages

File: docs/Modules/multichain.mdx (hunk 30 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:54]: I think current version it's simple enough. can we just keep it?

File: docs/Modules/multichain.mdx (hunk 29 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:40]: umm no :/

File: docs/Modules/multichain.mdx (hunk 28 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:48]: seems unncessary

File: docs/Modules/multichain.mdx (hunk 27 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx:23]: done

**Comment by livingrockrises** [comment]
> > > Typos and articles missing.
> > 
> > 
> > committed suggestions for typos.
> > which article?
> 
> https://www.grammarly.com/blog/articles/

oh ok I thought you meant like a piece of writing

File: docs/Modules/multichain.mdx (hunk 2 above)
Comment (livingrockrises) [reply docs/Modules/multichain.mdx]: done

File: docs/Modules/multichain.mdx (hunk 31 above)
Comment (himanshugarg06) [reply docs/Modules/multichain.mdx:58]: @livingrockrises in the docusaurus.config.js you can add the following 
 prism: {
        theme: lightCodeTheme,
        darkTheme: darkCodeTheme,
        **additionalLanguages: ['solidity']**
      },
 and use ```sol ``` it works. also can we shorten this method by moving some params to function line.

**Review by Aboudjem — DISMISSED** [review]

**Review by himanshugarg06 — APPROVED** [review]

**Review by Aboudjem — APPROVED** [review]
